packages/blob-propagator/src/utils.ts

```diff
@@ -0,0 +1,85 @@
+import fs from "fs";
+
+import { getBlobStorageManager } from "@blobscan/blob-storage-manager";
+import type { $Enums } from "@blobscan/db";
+
+import type { Blob } from "./types";
+
+export function checkBlobDataFileExists(versionedHash: string) {
+  const path = buildBlobDataFilePath(versionedHash);
+
+  return fs.promises
+    .access(path, fs.constants.F_OK)
+    .then(() => true)
+    .catch(() => false);
+}
+
+export function buildBlobDataFilePath(versionedHash: string) {
+  return `${versionedHash}.txt`;
+}
+
+export async function createBlobDataFile({ data, versionedHash }: Blob) {
+  const blobfilePath = buildBlobDataFilePath(versionedHash);
+
+  try {
+    if (await checkBlobDataFileExists(blobfilePath)) {
```

Comment:
Note we were calling the checkBlobDataFileExists with a path instead of the versionedHash.

Reply:
Oops, I refactor this part of the code to build the path in the check method itself and forgot to remove it from other places